commands/publish/lib/get-unpublished-packages.js

```diff
@@ -0,0 +1,41 @@
+"use strict";
+
+const fetch = require("npm-registry-fetch");
+const log = require("npmlog");
+const pReduce = require("p-reduce");
+
+module.exports = getUnpublishedPackages;
+
+// Only the abbreviated package metadata is needed to
+// determine which versions have been published. This
+// saves transfer time for packages with a lot of
+// history.
+const registryOptions = {
+  Accept: "application/vnd.npm.install-v1+json",
+};
+
+function getUnpublishedPackages(project, opts) {
+  log.silly("getPackageVersions");
+
+  let chain = Promise.resolve();
+
+  const mapper = (unpublished, pkg) =>
+    fetch.json(`/${pkg.name}`, Object.assign({}, opts, registryOptions)).then(
```

Comment:
You can't pass the configuration object to the `npm-registry-fetch` package. In my examples it will always fallback to NPM registry.

Reply:
It's because the config object isn't compatible with `Object.assign` you need to use
```js
Object.assign({}, opts.snapshot, registryOptions)
```

Reply:
Using `pacote` instead will resolve this.